core/src/consensus.rs

```diff
@@ -418,9 +489,21 @@ impl Tower {
         epoch_vote_accounts: &HashMap<Pubkey, (u64, Account)>,
     ) -> SwitchForkDecision {
         let root = self.lockouts.root_slot.unwrap_or(0);
+
         self.last_voted_slot()
             .map(|last_voted_slot| {
-                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap();
+                let last_vote_ancestors =
+                    ancestors.get(&last_voted_slot).unwrap_or_else(|| {
+                        if !self.is_stray_last_vote() {
+                            panic!("no ancestors found with slot: {}", last_voted_slot);
```

Comment:
hmmmm this will panic if our last vote was on some fork A, and then we start from a snapshot of another fork B right? I.e. in this example: https://github.com/solana-labs/solana/pull/10718#discussion_r474468319

If the last vote was on slot `10`, then we start from a snapshot `2`, we won't have the ancestors of `10` in `ancestors`, so this seems like it will panic.